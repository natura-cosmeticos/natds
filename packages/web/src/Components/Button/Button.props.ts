import { ButtonProps } from '@material-ui/core/Button'
import { PropTypes } from '@material-ui/core'
import * as React from 'react'

export type ButtonColor = PropTypes.Color;
export type ButtonSize = 'small' | 'medium' | 'large';
export type ButtonVariant = 'text' | 'outlined' | 'contained';

export interface IButtonProps extends Omit<ButtonProps, 'translate'> {

  /**
   * The content of the button
   *
   * @optional
   * @type node
   */
  children?: ButtonProps['children']

  /**
   * Override or extend the styles applied to the component.
   *
   * See [Material Button CSS API](https://material-ui.com/api/button/#css) for more details.
   *
   * @optional
   * @see https://material-ui.com/api/button/#css
   * @type object
   */
  classes?: ButtonProps['classes']

  /**
   * The color of the `Button` component. It supports those theme colors that make sense for this component.
   *
   * @default "default"
   * @optional
   * @type "default" | "inherit" | "primary" | "secondary"
   */
  color?: ButtonProps['color']

  /**
   * The component used for the root node. Either a `string` to use a DOM element or a component.
   *
   * @default "button"
   * @optional
   * @type element
   */
  component?: React.ReactElement

  /**
   *
   * If `true`, the button will be disabled.
   *
   * @default false
   * @type bool
   */
  disabled?: ButtonProps['disabled']

  textTransform?:'uppercase' | 'lowercase' | 'capitalize'

  /**
   * If `true`, no elevation is used.
   *
   * @default false
   * @optional
   * @type bool
   */
  disableElevation?: ButtonProps['disableElevation']

  /**
   * If `true`, the keyboard focus ripple will be disabled. `disableRipple` must also be `true`.
   *
   * @default false
   * @optional
   * @type bool
   */
  disableFocusRipple?: ButtonProps['disableFocusRipple']

  /**
   * If `true`, the ripple effect will be disabled.
   *
   * ⚠️ Without a ripple there is no styling for :focus-visible by default. Be sure
   * to highlight the element by applying separate styles with the `focusVisibleClassName`.
   *
   * @optional
   * @type bool
   */
  disableRipple?: ButtonProps['disableRipple']

  /**
   * Element placed after the children.
   *
   * @optional
   * @type node
   */
  endIcon?: ButtonProps['endIcon']

  /**
   * If `true`, the `Button` will take up the full width of its container.
   *
   * @default false
   * @optional
   * @type bool
   */
  fullWidth?: ButtonProps['fullWidth'];

  /**
   * The URL to link to when the button is clicked.
   * If defined, an `a` element will be used as the root node.
   *
   * @optional
   * @type string
   */
  href?: ButtonProps['href']

  /**
   * The size of the button.
   * `small` is equivalent to the dense button styling.
   *
   * @todo check if "small" size is still on UI Specs
   * @type "small" | "medium" | "large"
   */
  size?: ButtonProps['size']

  /**
   * Element placed before the children.
   *
   * @optional
   * @type node
   */
  startIcon?: ButtonProps['startIcon']

  /**
   * The variant to use.
   *
   * @optional
   * @type "contained" | "outlined" | "text"
   */
  variant?: ButtonProps['variant']
}
