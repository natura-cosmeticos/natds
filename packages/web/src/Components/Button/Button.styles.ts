import makeStyles from '@material-ui/core/styles/makeStyles'
import createStyles from '@material-ui/core/styles/createStyles'
import { IThemeWeb } from '../../Themes'

type ButtonRootStyles = {
  paddingLeft?: number,
  paddingRight?: number,
}

type ButtonLabelStyles = {
  lineHeight?: string,
  height?: number,
  marginBottom?: number,
  marginTop?: number
};

type ButtonStyles = {
  root: ButtonRootStyles,
  label: ButtonLabelStyles
}

export const getRootStyles: (theme: IThemeWeb) => (ButtonRootStyles) = (theme) => {
  if (typeof theme === 'undefined' || typeof theme.sizes === 'undefined' || typeof theme.sizes.small === 'undefined') {
    return {} as ButtonRootStyles
  }

  return {
    paddingLeft: theme?.sizes?.small,
    paddingRight: theme?.sizes?.small
  } as ButtonRootStyles
}

export const getLabelStyles: (theme: IThemeWeb) => (ButtonLabelStyles) = (theme) => {
  if (typeof theme === 'undefined' || typeof theme.sizes === 'undefined') {
    return {} as ButtonLabelStyles
  }

  return {
    lineHeight: `${theme.sizes.standard}px`,
    height: theme.sizes?.standard,
    marginBottom: theme.sizes.tiny,
    marginTop: theme.sizes.tiny
  } as ButtonLabelStyles
}

export const styles: (theme: IThemeWeb) => ButtonStyles = (theme) => ({
  root: getRootStyles(theme),
  label: getLabelStyles(theme)
} as ButtonStyles)

export const useStyles = makeStyles(createStyles(styles), { name: 'NatDSButton' })

export default useStyles
