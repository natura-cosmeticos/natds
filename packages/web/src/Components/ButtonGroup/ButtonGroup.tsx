import * as React from 'react'
import MaterialButtonGroup from '@material-ui/core/ButtonGroup'
import { IButtonGroupProps } from './ButtonGroup.props'

export { IButtonGroupProps } from './ButtonGroup.props'

/**
 * For more advanced usages, check [Material UI ButtonGroup docs](https://material-ui.com/components/buttons/#grouped-buttons)
 *
 * ## Import
 *
 * ```
 * import { Button, ButtonGroup } from '@naturacosmeticos/natds-web';
 * ```
 */
export const ButtonGroup = React.forwardRef<HTMLDivElement, IButtonGroupProps>((
  props: IButtonGroupProps, ref
) => {
  const { disableElevation, ...otherProps } = props

  return <MaterialButtonGroup {...otherProps} disableElevation={disableElevation} ref={ref} />
})

ButtonGroup.displayName = 'ButtonGroup'

export default ButtonGroup
