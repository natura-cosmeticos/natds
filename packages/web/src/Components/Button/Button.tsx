import * as React from 'react'

import MaterialButton from '@material-ui/core/Button'
import { IButtonProps } from './Button.props'
import useStyles from './Button.styles'

export { IButtonProps } from './Button.props'

/**
 * ```
 * import { Button } from '@naturacosmeticos/natds-web';
 * ```
 *
 * For more advanced usages, check [Material UI Button docs](https://material-ui.com/components/buttons/)
 */
export const Button = React.forwardRef<HTMLButtonElement, IButtonProps>(
  (props: IButtonProps, ref) => {
    const { classes, disableElevation, ...otherProps } = props

    const mergedClasses = {
      ...classes,
      ...useStyles()
    }

    return (
      <MaterialButton
        classes={mergedClasses}
        disableElevation={disableElevation}
        ref={ref}
        {...otherProps}
      />
    )
  }
)

Button.displayName = 'Button'
export default Button
