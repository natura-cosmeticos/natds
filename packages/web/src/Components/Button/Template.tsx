import { Story } from '@storybook/react'
import * as React from 'react'
import { IButtonProps } from './Button.props'
import { Button } from './Button'

export const Template : Story<IButtonProps> = (args) => <Button {...args} />

export const TemplateWithColors : Story<IButtonProps> = (args) => (
  <>
    <Button {...args} color="primary">Primary color</Button>
&nbsp;
    <Button {...args} color="secondary">Secondary color</Button>
&nbsp;
    <Button {...args} color="default">Default color</Button>
&nbsp;
  </>
)

export const TemplateWithSizes : Story<IButtonProps> = (args) => (
  <>
    <Button {...args} size="small">Small size</Button>
&nbsp;
    <Button {...args} size="medium">Medium size</Button>
&nbsp;
    <Button {...args} size="large">Large size</Button>
&nbsp;
  </>
)

export default Template
