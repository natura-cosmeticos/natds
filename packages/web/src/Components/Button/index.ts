import Button from './Button'

export { IButtonProps } from './Button.props'
export { Button }
export default Button
