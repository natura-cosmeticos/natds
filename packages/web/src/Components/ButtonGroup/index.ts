import ButtonGroup from './ButtonGroup'

export { IButtonGroupProps } from './ButtonGroup.props'
export { ButtonGroup }
export default ButtonGroup
