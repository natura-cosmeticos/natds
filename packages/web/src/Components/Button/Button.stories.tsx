import * as React from 'react'
import { Story } from '@storybook/react'
import { Button, IButtonProps } from './Button'
import { Icon } from '../Icon'
import { colors } from './__fixtures__/colors'
import { sizes } from './__fixtures__/sizes'
import { variants } from './__fixtures__/variants'
import { Template, TemplateWithColors, TemplateWithSizes } from './Template'

export default { component: Button, title: 'Components/Button' }

export const Playground : Story<IButtonProps> = Template.bind({})
Playground.args = {
  children: 'Button component text',
  color: colors.primary,
  endIcon: false,
  size: sizes.medium,
  startIcon: false,
  variant: variants.contained
}

export const ContainedVariant : Story<IButtonProps> = Template.bind({})
ContainedVariant.args = {
  ...Playground.args,
  variant: 'contained'
}

export const OutlinedVariant : Story<IButtonProps> = Template.bind({})
OutlinedVariant.args = {
  ...Playground.args,
  variant: 'outlined'
}

export const TextVariant : Story<IButtonProps> = Template.bind({})
TextVariant.args = {
  ...Playground.args,
  variant: 'text'
}

export const Disabled : Story<IButtonProps> = Template.bind({})
Disabled.args = {
  ...Playground.args,
  disabled: true
}

export const StartIcon : Story<IButtonProps> = Template.bind({})
StartIcon.args = {
  ...Playground.args,
  startIcon: <Icon name="outlined-action-favorite" />
}

export const EndIcon : Story<IButtonProps> = Template.bind({})
EndIcon.args = {
  ...Playground.args,
  endIcon: <Icon name="outlined-action-favorite" />
}

export const Colors : Story<IButtonProps> = TemplateWithColors.bind({})
Colors.args = Playground.args

export const InheritColor : Story<IButtonProps> = Template.bind({})
InheritColor.args = {
  ...Playground.args,
  color: colors.inherit
}
InheritColor.parameters = {
  docs: {
    description: {
      story: "The `inherit` color is useful when you need to apply a button to an `AppBar`'s `Toolbar`."
    }
  }
}

export const ForSnackbar : Story<IButtonProps> = Template.bind({})
ForSnackbar.args = {
  ...Playground.args,
  color: colors.inherit,
  variant: variants.text
}

export const ForDialogActions : Story<IButtonProps> = Template.bind({})
ForDialogActions.args = {
  children: 'Action text',
  color: colors.primary,
  variant: variants.text
}

export const Sizes : Story<IButtonProps> = TemplateWithSizes.bind({})
Sizes.args = Playground.args

/**
 * @todo Check if small size is still part of UI specs
 */
export const SmallSize : Story<IButtonProps> = Template.bind({})
SmallSize.args = {
  ...Playground.args,
  size: sizes.small
}

export const MediumSize : Story<IButtonProps> = Template.bind({})
MediumSize.args = {
  ...Playground.args,
  size: sizes.medium
}

export const LargeSize : Story<IButtonProps> = Template.bind({})
LargeSize.args = {
  ...Playground.args,
  size: sizes.large
}

export const FullWidth : Story<IButtonProps> = TemplateWithSizes.bind({})
FullWidth.args = {
  ...Playground.args,
  fullWidth: true
}

export const ForActions : Story<IButtonProps> = TemplateWithSizes.bind({})
ForActions.args = {
  children: 'Button label',
  size: sizes.small
}
