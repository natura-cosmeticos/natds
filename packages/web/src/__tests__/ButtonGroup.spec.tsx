import React from 'react';
import { expect } from 'chai';
import renderer from 'react-test-renderer';

import ButtonGroup from '../Components/ButtonGroup';
import Button from '../Components/Button';

describe('ButtonGroup component', () => {
  describe('rendering sizes', () => {
    test('should match snapshot - Size Small', () => {
      const component = renderer.create(<ButtonGroup size="small"><Button>One</Button><Button>Two</Button></ButtonGroup>).toJSON();
      expect(component).matchSnapshot('Small ButtonGroup snapshot');
    });

    test('should match snapshot - Size Medium', () => {
      const component = renderer.create(<ButtonGroup size="medium"><Button>One</Button><Button>Two</Button></ButtonGroup>).toJSON();
      expect(component).matchSnapshot('Medium ButtonGroup snapshot');
    });

    test('should match snapshot - Size Large', () => {
      const component = renderer.create(<ButtonGroup size="large"><Button>One</Button><Button>Two</Button></ButtonGroup>).toJSON();
      expect(component).matchSnapshot('Large ButtonGroup snapshot');
    });
  });

  describe('rendering variants', () => {
    test('should match snapshot - Low emphasis button (variant text)', () => {
      const component = renderer.create(<ButtonGroup variant="text"><Button>One</Button><Button>Two</Button></ButtonGroup>).toJSON();

      expect(component).to.matchSnapshot('Low emphasis ButtonGroup snapshot');
    });

    test('should match snapshot - Medium emphasis button (variant outlined)', () => {
      const component = renderer.create(<ButtonGroup variant="outlined"><Button>One</Button><Button>Two</Button></ButtonGroup>).toJSON();

      expect(component).to.matchSnapshot('Medium emphasis ButtonGroup snapshot');
    });

    test('should match snapshot - High emphasis button (variant contained)', () => {
      const component = renderer.create(<ButtonGroup variant="contained"><Button>One</Button><Button>Two</Button></ButtonGroup>).toJSON();

      expect(component).to.matchSnapshot('High emphasis ButtonGroup snapshot');
    });
  });
});
