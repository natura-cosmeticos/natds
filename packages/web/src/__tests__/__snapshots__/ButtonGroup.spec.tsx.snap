// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup component rendering sizes should match snapshot - Size Large: Medium ButtonGroup snapshot 1`] = `
<div
  className="MuiButtonGroup-root"
  role="group"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlined MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      One
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlined MuiButton-outlinedSizeLarge MuiButton-sizeLarge"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Two
    </span>
  </button>
</div>
`;

exports[`ButtonGroup component rendering sizes should match snapshot - Size Medium: Medium ButtonGroup snapshot 1`] = `
<div
  className="MuiButtonGroup-root"
  role="group"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlined"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      One
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlined"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Two
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`ButtonGroup component rendering sizes should match snapshot - Size Small: Small ButtonGroup snapshot 1`] = `
<div
  className="MuiButtonGroup-root"
  role="group"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      One
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Two
    </span>
  </button>
</div>
`;

exports[`ButtonGroup component rendering variants should match snapshot - High emphasis button (variant contained): High emphasis ButtonGroup snapshot 1`] = `
<div
  className="MuiButtonGroup-root MuiButtonGroup-contained"
  role="group"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedContained MuiButtonGroup-groupedContained"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      One
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedContained MuiButtonGroup-groupedContained"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Two
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`ButtonGroup component rendering variants should match snapshot - Low emphasis button (variant text): Low emphasis ButtonGroup snapshot 1`] = `
<div
  className="MuiButtonGroup-root"
  role="group"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedText MuiButtonGroup-groupedText"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      One
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButtonGroup-grouped MuiButtonGroup-groupedText MuiButtonGroup-groupedText"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Two
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;

exports[`ButtonGroup component rendering variants should match snapshot - Medium emphasis button (variant outlined): Medium emphasis ButtonGroup snapshot 1`] = `
<div
  className="MuiButtonGroup-root"
  role="group"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlined"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      One
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButtonGroup-grouped MuiButtonGroup-groupedOutlined MuiButtonGroup-groupedOutlined"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Two
    </span>
  </button>
</div>
`;
